### PR TITLE
8249627: Degrade Thread.suspend and Thread.resume

### PR DESCRIPTION
**Pull request by AlanBateman**
Degrade Thread.suspend/resume to throw UOE unconditionally.

Another step in the removal of this deadlock prone mis-feature from the user-facing API. Thread.suspend/resume have been deprecated since JDK 1.2 (1998) and terminally deprecated since Java 14. ThreadGroup.suspend/resume were degraded to throw UOE in Java 19. As of Java 19, Thread.suspend/resume continues to work for platform threads but throws UOE for virtual threads. The next step is to degrade both methods to throw UOE for all threads. A corpus search of 19M classes in 113k JAR files found only 22 classes using these methods so this change is unlikely to be disruptive.

The change requires some minor adjustments to the JVM TI and JDWP specifications, and a minor update to the JDI docs.

Leonid Mesnik is working on [PR10351](https://github.com/openjdk/jdk/pull/10351) to remove/replace the last few usages of Thread.suspend/resume from the hotspot tests (most of these can use JVMTI SuspendThread/ResumeThread).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8249627](https://bugs.openjdk.org/browse/JDK-8249627): Degrade Thread.suspend and Thread.resume
 * [JDK-8293975](https://bugs.openjdk.org/browse/JDK-8293975): Degrade Thread.suspend and Thread.resume (**CSR**)

### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**) ⚠️ Review applies to [d78ed990](https://git.openjdk.org/jdk/pull/10324/files/d78ed9901fa2742955c6627c6cb755c53af917ca)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**) ⚠️ Review applies to [d35954e6](https://git.openjdk.org/jdk/pull/10324/files/d35954e6be1e9140465ab969635e33467e1d06f7)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**) ⚠️ Review applies to [d78ed990](https://git.openjdk.org/jdk/pull/10324/files/d78ed9901fa2742955c6627c6cb755c53af917ca)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10324/head:pull/10324` \
`$ git checkout pull/10324`

Update a local copy of the PR: \
`$ git checkout pull/10324` \
`$ git pull https://git.openjdk.org/jdk pull/10324/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10324`

View PR using the GUI difftool: \
`$ git pr show -t 10324`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10324.diff">https://git.openjdk.org/jdk/pull/10324.diff</a>

</details>

### CODE REVIEW

**Review by plummercj — APPROVED** [review]
Changes look good.

**Review by sspitsyn — APPROVED** [review]
Looks good.
Thanks,
Serguei

**Comment by jaikiran** [comment]
Hello Alan, this PR deletes the `CountStackFrames.java` test case. Is that intentional? Looking at that test case it doesn't have any `suspend`/`resume` related testing and instead is just verifying that `Thread.countStackFrames()` throws the expected `UnsupportedOperationException`.

**Review by dholmes-ora — APPROVED** [review]
One minor nit but otherwise looks good!

Thanks.

File: src/java.base/share/classes/java/lang/SecurityManager.java [hunk 1]
```diff
@@ -494,7 +494,6 @@ private static ThreadGroup getRootGroup() {
      * calling thread is not allowed to modify the thread argument.
      * <p>
      * This method is invoked for the current security manager by the
-     * {@code stop}, {@code suspend}, {@code resume},
```
[the remark this answers is not in the document]
Comment (AlanBateman) [reply src/java.base/share/classes/java/lang/SecurityManager.java:497]: > Why is stop being removed in this PR?

It's just the sequencing of the two PRs and avoid merge conflicts. The Thread.stop should really edit this sentence, then this PR will edit it again. It was simpler to just do the SM edits in one PR.

**Review by sspitsyn — APPROVED** [review]

**Comment by AlanBateman** [comment]
Leonid has integrated JDK-8289607 to remove the last usages of Thread.suspend/resume from the hotspot tests so I've refreshed this PR to drop the temporary excluding of these tests.

**Review by dholmes-ora — APPROVED** [review]

**Comment by AlanBateman** [comment]
Wakeup bot

**Pull request closed without merging**